app/javascript/components/conditions/ConditionSetFormField/ConditionFormField/model.js

```diff
@@ -56,6 +62,31 @@ class ConditionModel {
         }
       },
     );
+
+    // Update rightQingOptions based on the selected leftQing, according to these rules:
+    // - Don't show selected leftQing in rightQingOptions
+    // - Allow only these type pairs:
+    //   - textual type -> textual type
+    //   - numeric type -> numeric type
+    //   - select_multiple -> none (literals only, due to ODK restriction)
+    //   - exact match for all other question types (qtypes must be identical)
+    reaction(
+      () => this.leftQingId,
+      (leftQingId) => {
+        if (leftQingId) {
+          const leftQing = this.refableQings.find((qing) => qing.id == leftQingId);
```

Comment:
Various lint errors, I'll fix up briefly after this review

Reply:
Thanks.